--- a/net/minecraft/world/entity/animal/frog/ShootTongue.java
+++ b/net/minecraft/world/entity/animal/frog/ShootTongue.java
@@ -85,7 +89,7 @@
             if (entity.isAlive()) {
                 frog.doHurtTarget(entity);
                 if (!entity.isAlive()) {
-                    entity.remove(Entity.RemovalReason.KILLED);
+                    entity.remove(Entity.RemovalReason.KILLED, EntityRemoveEvent.Cause.DEATH); // CraftBukkit - add Bukkit remove cause
                 }
             }
         }