--- a/net/minecraft/util/datafix/DataConverterRegistry.java
+++ b/net/minecraft/util/datafix/DataConverterRegistry.java
@@ -484,6 +484,18 @@
         datafixerbuilder.addFixer(new DataConverterItemFrame(schema44, false));
         Schema schema45 = datafixerbuilder.addSchema(1458, DataConverterRegistry.SAME_NAMESPACED);
 
+        // CraftBukkit start
+        datafixerbuilder.addFixer(new com.mojang.datafixers.DataFix(schema45, false) {
+            @Override
+            protected com.mojang.datafixers.TypeRewriteRule makeRule() {
+                return this.fixTypeEverywhereTyped("Player CustomName", this.getInputSchema().getType(DataConverterTypes.PLAYER), (typed) -> {
+                    return typed.update(DSL.remainderFinder(), (dynamic) -> {
+                        return DataConverterCustomNameEntity.fixTagCustomName(dynamic);
+                    });
+                });
+            }
+        });
+        // CraftBukkit end
         datafixerbuilder.addFixer(new DataConverterCustomNameEntity(schema45, false));
         datafixerbuilder.addFixer(new DataConverterCustomNameItem(schema45, false));
         datafixerbuilder.addFixer(new DataConverterCustomNameTile(schema45, false));